.github/workflows/scripts/code_coverage_disable_check.py

```
@@ -0,0 +1,177 @@
+File {file_path} contains code coverage disable statement."""
+                            )
+                            code_coverage_found = True
+        elif os.path.isfile(item):
+            # If it's a file, check it directly
+            if (
+                item.endswith(".tsx")
+                or item.endswith(".ts")
+                and not item.endswith(".test.tsx")
+                and not item.endswith(".test.ts")
+                and not item.endswith(".spec.tsx")
+                and not item.endswith(".spec.ts")
```

Ruff (0.14.10)

Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)